File: fetch-movie-info/src/main/java/com/getinfo/fetch_movie_info/service/MovieService.java
package com.getinfo.fetch_movie_info.service;

import com.getinfo.fetch_movie_info.entity.Movie;
import com.getinfo.fetch_movie_info.repository.MovieRepository;
import org.springframework.stereotype.Service;

import java.util.logging.Logger;

@Service
public class MovieService {
    private final Logger logger = Logger.getLogger(MovieService.class.getName());
    private final MovieRepository movieRepository;

    public MovieService(MovieRepository movieRepository){
        this.movieRepository = movieRepository;
    }

    public void saveMovie(Movie movie){
        movieRepository.save(movie);
        logger.info(String.format("Movie with title %s has been saved", movie.title));

    }

}

File: FetchInfoService/src/main/kotlin/service/fetch/info/service/MovieService.kt
package service.fetch.info.service

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import service.fetch.info.entity.Movie
import service.fetch.info.repository.MovieRepository

@Service
class MovieService @Autowired constructor(
    private val movieRepository: MovieRepository
    ) {

    fun saveMovie(movie: Movie){
        movieRepository.save(movie);
    }
}